src/map.h

```diff
@@ -1,72 +1,87 @@
-/*
- * rv32emu is freely redistributable under the MIT License. See the file
- * "LICENSE" for information on usage and redistribution of this file.
+#pragma once
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+/* C macro implementation of left-leaning 2-3 red-black trees.  Parent
```

Comment:
It is no longer a matter of macros. You should refine the comments and mention the origin (jemalloc).